Modules/socketmodule.c

```diff
@@ -7033,16 +7033,37 @@ os_init(void)
 }
 #endif
 
+static void
+sock_free_api(PySocketModule_APIObject *capi)
+{
+    Py_DECREF(capi->Sock_Type);
+    Py_DECREF(capi->error);
+    Py_DECREF(capi->timeout_error);
+    PyMem_Free(capi);
+}
 
-/* C API table - always add new things to the end for binary
-   compatibility. */
-static
-PySocketModule_APIObject PySocketModuleAPI =
+static void
+sock_destroy_api(PyObject *capsule)
 {
-    &sock_type,
-    NULL,
-    NULL
-};
+    void *capi = PyCapsule_GetPointer(capsule, PySocket_CAPSULE_NAME);
+    sock_free_api(capi);
```

Comment:
IMHO， calling `Py_DECREF()` and `PyMem_Free()` in here is not a big problem.
there have no code problem, so it's depend on you.

Reply:
Oh, I saw your code line #7158, pls forget this comment.